### PR TITLE
Support passing query params to launchDevTools

### PR DESCRIPTION
**Pull request by DanTup**
This allows passing args when calling the `launchDevTools` service. The args to go in the query string are in a field named `queryParams` so that if we want to add other arguments (to affect behaviour, not go on the querystring) we don't break it.

```js
{
	queryParams: {
		hide: "debugger",
		theme: config.useDevToolsDarkTheme ? "dark" : null,
	},
}
```

### CODE REVIEW

File: packages/devtools_server/lib/src/server.dart [hunk 1]
```diff
@@ -156,24 +156,33 @@ void serveDevTools({
...
-      final url = '$devToolsUrl/?uri=${Uri.encodeComponent(uri.toString())}';
+    service.registerServiceCallback(launchDevToolsService, (params) async {
+      final uriParams = <String, dynamic>{};
+
+      // Copy over queryParams passed by the client
+      if (params != null) {
+        params['queryParams']?.forEach((key, value) => uriParams[key] = value);
+      }
+
+      // Plus the URI to the VM service
```
Comment (kenzieschmoll) [inline packages/devtools_server/lib/src/server.dart]: nit: maybe rephrase this comment to be "Add the VM service uri param". I was confused by this wording initially.

**Comment by DanTup** [comment]
> General question - why aren't we getting the hide/theme/etc. params from stdin like we are getting the vm service uri?

These params are specific to launching DevTools in the browser, whereas the stdin stuff is just "register this VM service" (which will be used for things other than launching browsers in future), so it felt like it belonged with the request to launch.

Also, this way if the user changes theme, it'll take effect if they re-launch the browser, without having to restart the debug session.

File: packages/devtools_server/lib/src/server.dart (hunk 1 above)
Comment (DanTup) [reply packages/devtools_server/lib/src/server.dart]: Done!

It was written as a continuation of the comment above, but that probably wasn't a great idea.

I've also rebased on master, which I think will fix the broken build (I think it was the issue you solved yesterday).

**Comment by DanTup** [comment]
There's a better solution on the way, so for now we're just going to use a single hard-coded port and don't need this.

**Comment by DanTup** [comment]
Oops, wrong PR!

**Comment by DanTup** [comment]
Rebased on master due to a minor conflict in the `id`/`vmServiceUri` args in the launch-browser-tweaks land. Will land this if/when green.
